packages/list-plugins/lib/limiting/singleton.js

```diff
@@ -0,0 +1,43 @@
+const { composeHook, composeAccess } = require('../utils');
+
+exports.singleton = ({ preventDelete = true }) => ({
+  key,
+  keystone,
+  access,
+  hooks = {},
+  adminConfig = {},
+  ...rest
+}) => {
+  const newResolveInput = async ({ resolvedData, operation }) => {
+    if (operation === 'create') {
+      const list = keystone.getListByKey(key);
+      const query = `{${list.gqlNames.listQueryMetaName} { count }}`;
+      const {
+        data: { [list.gqlNames.listQueryMetaName]: listQuery } = {},
+        errors,
+      } = await keystone.executeQuery(query);
```

Comment:
I was actually going to use `action: { query }` parameter but realised that keystone is already available.  Should I change this to `query` parameter?